### DISCUSSION TITLE
Lag between skeleton predictions and video when displayed in GUI

### DISCUSSION THREAD

**Discussion by mjin1812** [talmolab/sleap#1221] [Help!] [answered]
I've been trialing sleap and I've been super happy so far with how easy labelling has been with such a user-friendly GUI. This is an awesome software!

However, after training (~only 40 frames) some demo videos of mice tested in the EPM, I've noticed a weird behavior with the displaying the skeleton predictions over my video after I open the generated 'predictions.slp' files in the GUI. It seems like there is a lag between where the skeleton is displayed and where the animal is as I progress through the video. 

However, when I export the skeleton overlay with the video using "Render Video Clip with Instances", there is no lag! At the link below, I've attached an example video I'm working with, the "predictions.slp" and "predictions.slp.mp4." 

[prediction data + video](https://drive.google.com/drive/folders/1Afdy_hWqhyAPvhPrueWkODNu3es0K7qn?usp=share_link)

I'm not sure if this is an issue anyone else has encountered or whether this is something specific to my setup. Any help correcting the lag would be appreciated!

Below is my setup info:

SLEAP: 1.3.0a0
TensorFlow: 2.8.4
Numpy: 1.21.5
Python: 3.7.16
OS: Windows-10-10.0.19041-SP0

**Comment by roomrys** [comment, marked as answer]
Hi @mjin1812,

Thanks for providing the video - we have been trying to recreate this problem ourselves, but have had trouble since it depends on both the video and the access pattern! The problem could be that the video is not [reliably seekable](http://blog.loopbio.com/video-io-1-introduction.html). Try [reencoding your video](https://sleap.ai/help.html#does-my-data-need-to-be-in-a-particular-format) and let us know how that works. Related #1115.

Thanks,
Liezl

**Reply by mjin1812** [reply]
Thank you so much for the fast response. I just tried predicting on a video that was re-encoded and everything works great!
